python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tracing.ipynb

```diff
@@ -0,0 +1,440 @@
+    "        Your only tool is search_tool - use it to find information.\n",
+    "        You make only one search call at a time.\n",
+    "        Once you have the results, you never do calculations based on them.\n",
+    "        \"\"\",\n",
+    "    )\n",
+    "\n",
+    "    data_analyst_agent = AssistantAgent(\n",
+    "        \"DataAnalystAgent\",\n",
+    "        description=\"An agent for performing calculations.\",\n",
+    "        model_client=model_client,\n",
+    "        tools=[percentage_change_tool],\n",
+    "        system_message=\"\"\"\n",
+    "        You are a data analyst.\n",
+    "        Given the tasks you have been assigned, you should analyze the data and provide results using the tools provided.\n",
+    "        If you have not seen the data, ask for it.\n",
+    "        \"\"\",\n",
+    "    )\n",
+    "\n",
+    "    text_mention_termination = TextMentionTermination(\"TERMINATE\")\n",
+    "    max_messages_termination = MaxMessageTermination(max_messages=25)\n",
+    "    termination = text_mention_termination | max_messages_termination\n",
+    "\n",
+    "    selector_prompt = \"\"\"Select an agent to perform task.\n",
+    "\n",
+    "    {roles}\n",
+    "\n",
+    "    Current conversation context:\n",
+    "    {history}\n",
+    "\n",
+    "    Read the above conversation, then select an agent from {participants} to perform the next task.\n",
+    "    Make sure the planner agent has assigned tasks before other agents start working.\n",
+    "    Only select one agent.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    task = \"Who was the Miami Heat player with the highest points in the 2006-2007 season, and what was the percentage change in his total rebounds between the 2007-2008 and 2008-2009 seasons?\"\n",
+    "    tracer_provider = configure_oltp_tracing()\n",
+    "    runtime = SingleThreadedAgentRuntime(tracer_provider=tracer_provider)\n",
+    "    tracer = trace.get_tracer(\"autogen-test-agentchat\")\n",
+    "    with tracer.start_as_current_span(\"runtime\"):\n",
+    "        runtime.start()\n",
+    "        team = SelectorGroupChat(\n",
+    "            [planning_agent, web_search_agent, data_analyst_agent],\n",
+    "            model_client=model_client,\n",
+    "            termination_condition=termination,\n",
+    "            selector_prompt=selector_prompt,\n",
+    "            allow_repeated_speaker=True,  # Allow an agent to speak multiple turns in a row.\n",
+    "            runtime=runtime,\n",
+    "        )\n",
+    "        await Console(team.run_stream(task=task))\n",
+    "\n",
+    "    await runtime.close()\n",
+    "    await model_client.close()\n",
+    "\n",
+    "\n",
+    "# asyncio.run(main())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Overriding of current TracerProvider is not allowed\n"
```

Comment:
You can see here is a warning. 

We can have a separate cell just for setting up the global trace provider without making it a function `configure_oltp_tracing`. 

The runtime now automatically gets the current default trace provider see:

https://github.com/microsoft/autogen/blob/46add11ec7304384496322f9c1f2ee0da21044f7/python/packages/autogen-core/src/autogen_core/_telemetry/_tracing.py#L27-L28

Reply:
thanks, updated. 
Given that distributed runtime is still [marked as experimental](Distributed tracing via grpc agent runtime.), perhaps we can add examples with `GrpcWorkerAgentRuntimeHost` and workers later? 
Thoughts?

Reply:
> we can add examples with GrpcWorkerAgentRuntimeHost and workers later?

Yes.